Fix liboat_hook.so inaccessible in some device

JingMatrix opened the pull request:
User has reported `dex2oat` failure with SELinux log:

W dex2oat64: type=1400 audit(0.0:922): avc: denied { read } for path="/data/adb/modules/zygisk_lsposed/bin/liboat_hook64.so" dev="dm-58" ino=91204 scontext=u:r:dex2oat:s0 tcontext=u:object_r:adb_data_file:s0 tclass=file permissive=0

JingMatrix committed:
Fix liboat_hook.so inaccessible in some device

User has reported `dex2oat` failure with SELinux log:

W dex2oat64: type=1400 audit(0.0:922): avc: denied { read } for path="/data/adb/modules/zygisk_lsposed/bin/liboat_hook64.so" dev="dm-58" ino=91204 scontext=u:r:dex2oat:s0 tcontext=u:object_r:adb_data_file:s0 tclass=file permissive=0

JingMatrix commented:
@aayushrautela try the latest CI build to see if your problem got fixed.

aviraxp commented:
I think the issue is init relabels xposed_file when doing restorecon on /data, which is usually performed during ota. That's why magisk_file is often used in that dir.
https://android.googlesource.com/platform/system/sepolicy/+/refs/heads/main/private/init.te#362
Try make xposed_file system_file_type.

JingMatrix committed:
Define xposed_file as system_file_type

The init process may relable SELinux context, which can be prevented by marking the context xposed_file with type system_file_type.

See https://cs.android.com/android/platform/superproject/main/+/main:system/sepolicy/private/init.te;l=335

JingMatrix commented:
Thanks for the information, I will let users to test your conjecture.

JingMatrix committed:
Set SELinux context of hookers in Dex2OatService
